
### Kinematic configuration parameter

# Debug mode
activate_output: true
activate_debug_output: true
activate_controller_node_output: true
plotting_result: true

# Clock frequency //hz
clock_frequency: 20

#Optimal control problem dimensions
state_dim: 4
control_dim: 2

#Inputs and output topics topic
output_cmd: '/prius'
robot_state_topic: '/base_pose_ground_truth'

# Kinematic chain parameter
robot_base_link: base_link
target_frame: map
tracking_frame: map

# Weight on the slack variable for collision collision
slack_weight: 10000
repulsive_weight: 5

# Parameters for collision constraints
obstacles:
      n_obstacles: 2
      sub_ellipse_topic: ellipse_objects_feed
      n_discs: 1
      ego_l: 0.5
      ego_w: 0.4

# predictive_config: # minimum_collision_distance should not be larger than ball radius
self_collision:
     ball_radius: 0.12
     minimum_collision_distance: 0.15
     collision_weight_factor: 0.01

     # the following obstacles are considered for collision avoidance
     collision_check_obstacles: []

constraints:
     velocity_constraints:
           min: [0]
           max: [13.8]
     #orientation_constraints:

acado_config:
     max_num_iteration: 70
     kkt_tolerance: 1e-3
     integrator_tolerance: 1e-3
     start_time_horizon: 0.0
     end_time_horizon: 5.0 # 5.0  # control and prediction horizon
     discretization_intervals: 50
     sampling_time: 0.1
     use_lagrange_term: false
     use_LSQ_term: false
     use_mayer_term: true
     weight_factors:
           lsq_state_weight_factors:
                #always 6 component 3 linear/position and 3 angular/Euler angle
                [10.0, 40.0, 3.0,3.0]
           lsq_control_weight_factors:
                # same as degree of freedom
                [2.0, 2.0]
           lsq_state_terminal_weight_factors:
                #always 6 component 3 linear/position and 3 angular/Euler angle
                [10.0, 40.0, 3.0,3.0]
